src/vector.js

```diff
@@ -2,6 +2,107 @@
 
 export var Vector = {};
 
+Vector.set = function (v) {
+    var V = [];
+    var lim = v.length;
+    for (var i = 0; i < lim; i++) {
+        V[i] = v[i];
+    }
+    return V;
+};
+
+Vector.neg = function (v) {
+    var V = [];
+    var lim = v.length;
+    for (var i = 0; i < lim; i++) {
+        V[i] = v[i] * -1;
+    }
+    return V;
+};
+
+// Addition of two vectors
+Vector.add = function (v1, v2) {
+    var v = [];
+    var lim = Math.min(v1.length,v2.length);
+    for (var i = 0; i < lim; i++) {
+        v[i] = v1[i] + v2[i];
+    }
+    return v;
+};
+
+// Substraction of two vectors
+Vector.sub = function (v1, v2) {
+    var v = [];
+    var lim = Math.min(v1.length,v2.length);
+
+    for (var i = 0; i < lim; i++) {
+        v[i] = v1[i] - v2[i];
+    }
+    return v;
+};
+
+// Multiplication of two vectors
+Vector.mult = function (v1, v2) {
+    var v = [],
+        len = v1.length,
+        i;
+    
+    if (typeof v2 === 'number') {
+        // Mulitply by scalar
+        for (i = 0; i < len; i++) {
+            v[i] = v1[i] * v2;
+        }
+    }
+    else {
+        // Multiply two vectors
+        len = Math.min(v1.length,v2.length);
+        for (i = 0; i < len; i++) {
+            v[i] = v1[i] * v2[i];
+        }
+    }
+    return v;
+};
+
+// Division of two vectors
+Vector.div = function (v1, v2) {
+    var v = [], 
+        i;
+    if(typeof v2 === 'number'){
+        // Divide by scalar
+        for (i = 0; i < v1.length; i++){
+            v[i] = v1[i] / v2;
+        }
+    } else {
+        // Divide to vectors
+        var len = Math.min(v1.length,v2.length);
+        for (i = 0; i < len; i++) {
+            v[i] = v1[i] / v2[i];
+        }
+    }
+    return v;
+};
+ 
+// Get 2D perpendicular FAST
+Vector.perp = function (v1, v2) {
+    return [ v2[1] - v1[1], 
+             v1[0] - v2[0] ];
+};
+
+// Get 2D vector rotated 
+Vector.rot = function (v, a) {
```

Comment:
Objects are passed by reference, this is going to overwrite the contents of the argument vector `v`. Set `V = []` to make a new vector instead.